src/dataProvider/model/MediaComposition.js

```diff
@@ -0,0 +1,322 @@
+/**
+ * @class MediaComposition
+ */
+class MediaComposition {
+  /**
+   * Find a chapter by its URN.
+   *
+   * @param {String} urn
+   *
+   * @returns {Object} chapter
+   */
+  findChapterByUrn(urn) {
+    if (this.chapterList) {
+      const [chapter] = this.chapterList.filter(
+        (element) => element.urn === urn
+      );
+
+      return chapter;
+    }
+
+    return undefined;
+  }
+
+  /**
+   * Find all DRM vendors available for a resource.
+   *
+   * @param {Resource} resource
+   *
+   * @returns {Array} of DRM vendors
+   */
+  static findDrmListByResource(resource) {
+    return resource.drmList;
+  }
+
+  /**
+   * Return a segment from main chapter following segmentUrn in mediaComposition.
+   *
+   * @returns {Object|undefined} main segment
+   */
+  findMainSegment() {
+    if (!this.segmentUrn) {
+      return undefined;
+    }
+
+    const segmentList = this.getMainSegments();
+    const [segment] = segmentList.filter(
+      (element) => element.urn === this.segmentUrn
+    );
+
+    return segment;
+  }
+
+  /**
+   * Find resource list by URN.
+   *
+   * @param {String} urn
+   * @returns {Array|undefined} of resources
+   */
+  findResourceListByUrn(urn) {
+    const chapterByUrn = this.findChapterByUrn(urn);
+
+    if (chapterByUrn) {
+      return chapterByUrn.resourceList || [];
+    }
+
+    return undefined;
+  }
+
+  /**
+   * A list of chapters.
+   *
+   * @returns {Array} of chapters
+   */
+  getChapters() {
+    return this.chapterList;
+  }
```

Comment:
Not sure to understand the purpose of this method. Is it really useful?

Reply:
I'll tend to keep it, the main reason is it helps the code completion.

Reply:
You should use a better IDE then ;)

Reply:
I'm still waiting for `Xcode` to be available on Linux, it's the only one really worth using.